frontend/public/components/factory/table.tsx

```diff
@@ -424,13 +442,39 @@ type ComponentProps = {
   kindObj?: K8sResourceKindReference;
 };
 
+const getActiveColumns = (
+  Header: any,
+  componentProps: ComponentProps,
+  activeColumns: Set<string>,
+) => {
+  let columns = Header(componentProps);
+  if (_.isEmpty(activeColumns) && columns.find((col) => col.additional)) {
+    activeColumns = new Set(
+      columns.map((col) => {
+        if (col.id && !col.additional) {
+          return col.id;
+        }
+      }),
+    );
+  }
+  if (!_.isEmpty(activeColumns)) {
+    columns = columns?.filter((col) => activeColumns.has(col.id) || col.title === '');
+  }
+
+  const showNamespace = UIActions.getActiveNamespace() === ALL_NAMESPACES_KEY;
+  columns = columns.filter((column) => column.id !== 'namespace' || showNamespace);
```

Comment:
nit: you can skip iterating over the columns if `showNamespace` is true

```suggestion
  if (!showNamespace) {
    columns = columns.filter((column) => column.id !== 'namespace');
  }
```

Reply:
good point less churn.